builtin/logical/ad/backend_config_test.go

```diff
@@ -0,0 +1,235 @@
+package ad
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/builtin/logical/ad/config"
+	"github.com/hashicorp/vault/logical"
+)
+
+var (
+	ctx     = context.Background()
+	storage = &logical.InmemStorage{}
+)
+
+func TestConfigWriteReadDelete(t *testing.T) {
+
+	b, err := Factory(ctx, &logical.BackendConfig{
+		Logger:      hclog.NewNullLogger(),
+		StorageView: storage,
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Write
+	req := &logical.Request{
+		Operation: logical.UpdateOperation,
+		Path:      config.BackendPath,
+		Storage:   storage,
+		Data: map[string]interface{}{
+			"username":    "tester",
+			"password":    "pa$$w0rd",
+			"urls":        "ldap://138.91.247.105",
+			"certificate": validCertificate,
+			"dn":          "example,com",
+		},
+	}
+	resp, err := b.HandleRequest(ctx, req)
+	if err != nil {
+		t.Error(err)
+	}
+	verifyResponse(t, resp)
+	if !configIsStoredAsExpected() {
```

Comment:
I think it'd be nicer to change the helper to `configIsStoredAsExpected(t *testing.T)` + using `t.Helper()`, and then calling `configIsStoredAsExpected(t)` in here than to do this plus `t.FailNow()` since the failure output might not be verbose enough. For each of the cases in `configIsStoredAsExpected(t *testing.T)` that currently returns `false`, you can instead to a `t.Fatalf(...)` so that you know what check exactly failed.

Reply:
Totally! I'll update to that. I actually had it that way originally but then as the test changed here and there, stripped it out, but then now they way they ended up it would be better that way again.